libraries/AP_Terrain/TerrainIO.cpp

```diff
@@ -178,8 +178,7 @@ void AP_Terrain::open_file(void)
     // create directory if need be
     if (!directory_created) {
         *p = 0;
-        mkdir(file_path, 0755);
```

Comment:
I'm not sure, it should return something as it wants to be POSIX compliant, but I agree if it can silently fail then there is no point in this check (and we have a different problem at that point).